### ISSUE TITLE
Bug with throwing exception from void method using WithBDDMockito

### ISSUE THREAD

**Issue by nickheniser** [szpak/mockito-java8#7]
The following code will cause java.lang.NoSuchMethodError instead of actually throwing the exception.

```
@Mock BufferedWriter mockedBufferedWriter
willThrow(new IOException()).given(mockedBufferedWriter).write(anyString());
```

**Comment by szpak** [comment]
What Mockito and Mockito-Java8 versions do you use?

**Comment by szpak** [comment]
(as I'm not able to reproduce it with Mockito 2.0.44-beta and mockito-java8 2.0.0-beta)

**Comment by nickheniser** [comment]
I am using Mockito 2.042-beta and mockito-java8 2.0.0-beta.  But I also tried with 2.0.44 and it still is an issue.

```
package com.example;

import info.solidsoft.mockito.java8.api.WithBDDMockito;

import java.io.BufferedWriter;
import java.io.IOException;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

@RunWith(MockitoJUnitRunner.class)
public class ReproducableTest implements WithBDDMockito {

	@Mock BufferedWriter writer;

	@Test(expected = IOException.class)
	public void testWrongExceptionThrown() throws IOException {

		willThrow(new IOException()).given(writer)
			.write(anyString());

		writer.write("throws NoSuchMethodError instead of IOException");

	}
}

```

**Comment by szpak** [comment]
In the end I found a reason. `BDDMockito.willThrow` accepts `Throwable... toBeThrown` while my `WithBDDMockito.willThrow` was accepting just `Throwable`. I fixed it and will be available in the next version of mockito-java8.

Btw, your code works fine when put into the mockito-java8 project itself as a test. I was needed to create a separate project to reproduce that issue...

Thanks for your report.

**Comment by nickheniser** [comment]
Very odd about it working in the project and not externally.  Thanks for
making a fix, I will watch for the new release!

Nick

**Comment by szpak** [comment]
I don't like unexplained situation, so I look into the case and I had found that even with a little bit  different method signatures with varargs the compiler is able to generate valid byte code converting my:

```
return BDDMockito.willThrow(toBeThrown);
```

into

```
return BDDMockito.willThrow(new Throwable[]{toBeThrown});
```

I suspected enhancements in the compiler, but I wasn't able to reproduce it with very early Java 8 versions. In the end I have found that mockito-java8 2.0.0-beta.1 was built using mockito 2.0.22-beta as a dependency and in 2.0.32-beta the signatures was [changed](https://github.com/mockito/mockito/commit/6df356efcaf28262ec75db2d8e7a2bf7eafb5349#diff-445b8f49e35c2b3edc0960e21c7f54abR348). Therefore your test case works fine from the project itself (it uses 2.0.44-beta and compiler adjust the code) and the problem only occurs with the original 2.0.0-beta.1 jar built with old method signatures in Mockito . The new version would be probably incompatible with Mockito <2.0.32 (if changed methods are called).
